spec/controllers/articles_controller_spec.rb

```diff
@@ -0,0 +1,336 @@
+# frozen_string_literal: true
+
+require 'spec_helper'
+
+describe ArticlesController, type: :controller do
+  let(:user) { create :user, :role_article_meta }
+  let(:article_category_a) { create :article_category, name: "AAAA" }
+  let(:article_category_b) { create :article_category, name: "BBBB" }
+  let(:article_a) { create :article, name: 'AAAA', note: "AAAA", unit: '250 g', article_category: article_category_a, availability: false }
+  let(:article_b) { create :article, name: 'BBBB', note: "BBBB", unit: '500 g', article_category: article_category_b, availability: true }
+  let(:article_c) { create :article, name: 'CCCC', note: "CCCC", unit: '500 g', article_category: article_category_b, availability: true }
+
+  let(:supplier) { create :supplier, articles: [article_a, article_b] }
+  let(:order) { create :order }
+
+  before { login user }
+
+  describe 'GET index' do
+    it 'assigns sorting on articles' do
+      sortings = [
+        ['name', [article_a, article_b]],
+        ['name_reverse', [article_b, article_a]],
+        ['unit', [article_a, article_b]],
+        ['unit_reverse', [article_b, article_a]],
+        ['article_category', [article_a, article_b]],
+        ['article_category_reverse', [article_b, article_a]],
+        ['note', [article_a, article_b]],
+        ['note_reverse', [article_b, article_a]],
+        ['availability', [article_a, article_b]],
+        ['availability_reverse', [article_b, article_a]]
+      ]
+      sortings.each do |sorting|
```

Comment:
if all `_reverse` sorting has  the same expected results: how do you test that the code for `name_reverse` is does not sort by e.g. `unit`, which would lead to the same result. IMHO this test gives a false impression and should either done in a way that ensures that the correct part is tested (e.g. with more than two articles) or should be removed, to be clear about missing test coverage.